dxda.go

```diff
@@ -151,19 +151,27 @@ func makeRequestWithHeadersFail(requestType string, url string, headers map[stri
 		}
 	}
 
-	req, err := retryablehttp.NewRequest(requestType, url, bytes.NewReader(data))
-	check(err)
-	for header, value := range headers {
-		req.Header.Set(header, value)
-	}
-	resp, err := client.Do(req)
-	check(err)
-	defer resp.Body.Close()
-	status = resp.Status
-	body, _ = ioutil.ReadAll(resp.Body)
+	// Perpetually retry on 503 (e.g. platform downtime/throttling)
+	for {
+		req, err := retryablehttp.NewRequest(requestType, url, bytes.NewReader(data))
+		check(err)
+		for header, value := range headers {
+			req.Header.Set(header, value)
+		}
+		resp, err := client.Do(req)
+		check(err)
+		defer resp.Body.Close()
```

Comment:
I don't know what happens to this defer statement if there is, indeed, a 503. Is it only released at function return? If so, you could have a large accumulation of responses.

Reply:
Good point, let me modify it.